### PR TITLE
Improve README for storage client.

### LINKED ISSUE #2363: Update Storage README with library quality expectations
Hi,

The README for the Cloud Storage library [README.md](https://github.com/googleapis/google-cloud-cpp/blob/master/google/cloud/storage/README.md) must be updated with the following:

1. Links to Cloud Storage doc (cloud.google.com/storage/docs).
2. Links to C++ API reference documentation.
3. Links to the getting started documentation in the Reference API.
4. Includes a description of the Product API.
> Cloud Storage allows world-wide storage and retrieval of any amount of data at any time. You can use Cloud Storage for a range of scenarios including serving website content, storing data for archival and disaster recovery, or distributing large data objects to users via direct download.
5. Include information about Contributing, License, and Support.
6. Include information about what Versioning means in this library. 

Here's an example in [Ruby readme](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-storage/README.md), though it's using an outdated description of the product.

Per the library checklist (@coryan) has context.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #2363.

Please read go/cloud-cxx:gcs-versioning if you have questions about the versioning scheme.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2430)

### CODE REVIEW

File: google/cloud/storage/README.md [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
+* The [client library reference][doxygen-link] contains full documentation
+  for the Cloud Storage C++ client library, and includes code samples for all
+  APIs.
+* A [quick start][quickstart-link] is part of the client library reference.
+* Please consult the [Cloud Storage website][gcs-docs-link] for general
+  documentation on Cloud Storage features, APIs, other client libraries, etc.
+
+## Contributing changes
+
+See [`CONTRIBUTING.md`](../CONTRIBUTING.md) for details on how to contribute to
```
Comment (frankyn) [inline google/cloud/storage/README.md]: Not the correct path. Should be `../../../CONTRIBUTING.md`.

File: google/cloud/storage/README.md [hunk 2]
```diff
@@ -1,22 +1,46 @@
...
+this project, including how to build and test your changes as well as how to
+properly format your code.
+
+## Licensing
+
+Apache 2.0; see [`LICENSE`](../LICENSE) for details.
```
Comment (frankyn) [inline google/cloud/storage/README.md]: Path isn't correct here as well.

File: google/cloud/storage/README.md [hunk 3]
```diff
@@ -1,22 +1,46 @@
...
+[doxygen-shield]: https://img.shields.io/badge/documentation-master-brightgreen.svg
+[doxygen-link]: https://googleapis.github.io/google-cloud-cpp/latest/storage
+[quickstart-link]: https://googleapis.github.io/google-cloud-cpp/latest/storage/index.html
+[gcs-docs-link]: https://cloud.google.com/storage/docs
+
 ## Status
 
 This library support Google Cloud Storage at the
-[Beta](../README.md#versioning) quality level.
+[Beta](../README.md#versioning) quality level. Please note that, as is often the
```
Comment (frankyn) [inline google/cloud/storage/README.md]: Update path to point to root README.

File: google/cloud/storage/README.md (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/README.md]: Fixed.

File: google/cloud/storage/README.md (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/README.md]: Fixed.

File: google/cloud/storage/README.md (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/README.md]: Fixed.

**Comment by coryan** [comment]
PTAL. I also rebased to resolve conflicts.

**Pull request merged**